enhancements/agent-installer/automated-workflow.md

```diff
@@ -0,0 +1,491 @@
+    on the hypervisor host.
+11. The orchestration tool configures the host to boot the ISO. For
+    bare metal, that means configuring the baseboard management
+    controller (BMC) in each host to expose the ISO through the
+    virtual media interface. For vSphere or other hypervisors, similar
+    settings of the VM need to be adjusted.
+12. The orchestration tool boots the host.
+13. The orchestration tool uses `openshift-install wait-for
+    install-complete` to watch the deployment progress and wait for it
+    to complete.
+14. On each host, the startup script in the image selects the correct
+    network settings based on the MAC addresses visible and applies
+    the network settings for each interface.
+15. On node0, the startup script in the image recognizes that it is on
+    node0 and launches the assisted service and the create-cluster
+    service to drive the deployment.
+16. On node0, the create-cluster service waits for the assisted
+    service to start, then creates an InfraEnv using the API.
+17. On node0, the create-cluster service waits for all of the known
+    agents to be registered with the service.
+18. On all nodes (including node0), the startup script in the image
+    connects to the assisted service on node0 to find the InfraEnv ID
+    for the cluster being deployed (there will only be 1). After it
+    has the ID, the startup script launches the assisted agent with
+    that ID and configured to connect to the assisted service on
+    node0.
+19. All of the agents register their hosts with the assisted service.
+20. The assisted service performs its standard validations for all of the
+    hosts.
+21. On node0, the create-cluster service sees that all known agents
+    have registered and the hosts have passed validation. It uses the
+    assisted service API to trigger the deployment to proceed.
+22. The assisted service runs the OpenShift installer and performs the
+    cluster deployment in the usual way, except that the nodes will
+    use the ISOs they booted from to install RHCOS instead of fetching
+    a new one from the assisted service.
+23. The `openshift-install wait-for install-complete` process sees the
+    OpenShift API become available and starts using it to watch the
+    cluster complete deployment, combining the information provided by
+    `ClusterVersion` and `ClusterOperator` resources with the assisted
+    service progress API.
+24. On node0, when all of the other hosts have been deployed and
+    bootstrapping is complete, the agent reboots node0.
+25. The `openshift-install wait-for install-complete` process loses
+    the connection to the assisted service REST API and starts relying
+    entirely on the OpenShift API for data.
+26. The `openshift-install wait-for install-complete` process sees
+    that the `ClusterVersion` API in the cluster shows that the
+    deployment has completed, and reports success then exits.
```

Comment:
What if the installation fails at this point?  What if it failed before node0 rebooted?

Reply:
if it's before node0 reboots, we'd get the info from assisted-service. If it's after that, there'll need to be a timeout condition seeing that the cluster api does not list node0 as a master (obviously with SNO we get nothing at that point).